File: api/src/main/java/com/wbrawner/twigs/session/SessionCleanupTask.java
package com.wbrawner.twigs.session;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import java.util.Date;

@Component
public class SessionCleanupTask {
    private final UserSessionRepository sessionRepository;

    @Autowired
    public SessionCleanupTask(UserSessionRepository sessionRepository) {
        this.sessionRepository = sessionRepository;
    }

    @Scheduled(cron = "0 0 * * * *")
    public void cleanup() {
        sessionRepository.deleteAllByExpirationBefore(new Date());
    }
}

File: api/src/main/kotlin/com/wbrawner/budgetserver/session/SessionCleanupTask.kt
package com.wbrawner.budgetserver.session

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import java.util.*

@Component
open class SessionCleanupTask @Autowired constructor(private val sessionRepository: UserSessionRepository) {
    @Scheduled(cron = "0 0 * * * *")
    open fun cleanup() {
        sessionRepository.deleteAllByExpirationBefore(Date())
    }
}